### PR TITLE
Image Customizer: Partition UUID reset.

### PR DESCRIPTION
**Pull request by cwize1**
Add support for resetting the partition and filesystem UUIDs without needing to customize the partition layout.

In addition, this change has necessitated adding support for resetting the bootloader configuration even when the partition layout isn't customized.

---

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [x] Ready to merge

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Added UTs.

### CODE REVIEW

File: toolkit/tools/imagecustomizer/README.md [hunk 1]
```diff
@@ -46,7 +46,8 @@ Disadvantages:
 3. Install prerequisites: `qemu-img`, `rpm`, `dd`, `lsblk`, `losetup`, `sfdisk`,
```
Comment (gmileka) [inline toolkit/tools/imagecustomizer/README.md:46]: Chris, do we need to add docs for the new api surface and its dependencies?

Comment (cwize1) [reply toolkit/tools/imagecustomizer/README.md:46]: Good catch! Thanks. Fixed.

File: toolkit/tools/imagecustomizer/README.md [hunk 2]
```diff
@@ -46,7 +46,8 @@ Disadvantages:
...
    `udevadm`, `flock`, `blkid`, `openssl`, `sed`, `createrepo`, `mksquashfs`,
    `genisoimage`, `parted`, `mkfs`, `mkfs.ext4`, `mkfs.vfat`, `mkfs.xfs`, `fsck`,
-   `e2fsck`, `xfs_repair`, `resize2fs`, `zstd`, `veritysetup`.
+   `e2fsck`, `xfs_repair`, `resize2fs`, `tune2fs`, `xfs_admin`, `fatlabel`, `zstd`,
```
Comment (romoh) [inline toolkit/tools/imagecustomizer/README.md:49]: should those be added in the dockerfile too?

File: toolkit/tools/pkg/imagecustomizerlib/customizepartitionsuuids.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+			continue
+		}
+
+		newUuid, err := resetFileSystemUuid(partition)
+		if err != nil {
+			return fmt.Errorf("failed to reset partition's (%s) filesystem (%s) UUID:\n%w", partition.Path,
+				partition.FileSystemType, err)
+		}
+
+		newUuids[i] = newUuid
```
Comment (romoh) [inline toolkit/tools/pkg/imagecustomizerlib/customizepartitionsuuids.go:50]: should we assert that the uuid is set using `blkid` or so?

**Review by romoh — COMMENTED** [review]
left few comments

File: toolkit/tools/imagecustomizer/README.md (hunk 2 above)
Comment (cwize1) [reply toolkit/tools/imagecustomizer/README.md:49]: These tools are already included in the existing packages. Namely, `e2fsprogs`, `dosfstools`, and `xfsprogs`.

File: toolkit/tools/pkg/imagecustomizerlib/customizepartitionsuuids.go (hunk 3 above)
Comment (cwize1) [reply toolkit/tools/pkg/imagecustomizerlib/customizepartitionsuuids.go:50]: Probably not. The entries under /dev are only updated asynchronously. So, the new value might not show up immediately.

**Review by romoh — APPROVED** [review]

**Review by dmcilvaney — APPROVED** [review]

**Pull request merged**